### DISCUSSION TITLE
Is there any example about using Zitadel with microservice application?

### DISCUSSION THREAD

**Discussion by Lan-Phan** [zitadel/zitadel#6202] [Q&A]
Hello,

I have read through some key concepts of Zitadel, but still don't know exactly how I can apply it to our microservice application.
Do you guys have any microservice example using Zitadel as IdP for reference?

I'd like to show 1 example here but I still have some concerns about it:
* base microservice here: https://github.com/thangchung/go-coffeeshop + imaginary booking service for customer to book tables (totally it has 6 services: product, barista, kitchen, counter, web, booking. They have nice image showing its flow and service dependency)
* Imagine that we'll store info of USER and CUSTOMER for shop
* USER have 3 roles: Admin, Product (one have permissions to create/update/delete and manage list of products), CounterPerson (one have to input orders only)

With that setup, I think I'll config Zitadel as followings:
* 1 instance SHOP
* instance SHOP has 2 orgs USER and CUSTOMER to keep list of users and customers separated
* org CUSTOMER has:
    ++ list customers here
    ++ project 1 contains app booking + no role
* org USER has:
    ++ list users here
    ++ project 1 contains app (barista and kitchen) + role (Admin and CounterPerson)
    ++ project 2 contains app (product) + role (Admin and Product)
    ++ project 3 contains app (counter) + role (Admin and CounterPerson)
    ++ project 4 contains app (web) + role (Admin and Product and CounterPerson)

However, after reading your Concepts about Application, seems my configuration is not correct (my feeling) but I don't know how to create a correct config for this setup. I'm very appreciated for your help.

**Comment by hifabienne** [comment]
Hei @Lan-Phan 
I am not completely sure if I understand everything but here is how I would set it up:
org SHOP:
- project 1 
    - contains all apps (product, barista, kitchen, counter, web, booking) 
    - roles (Admin, CounterPerson and Product)
    - grant the project to the Customers organization (probably without a role, as customers do not need a role, or with the roles needed)
- users, I would add the employees here

org Customers:
- users for customers

Does that make sense to you?
